rotkehlchen/externalapis/defillama.py

```diff
@@ -86,6 +90,9 @@ def _query(
 
         return decoded_json
 
+    def is_punished(self) -> bool:
+        return ts_now() - self._punishment_info['last_punished'] <= ORACLE_PUNISHMENT_TS
```

Comment:
as an example I will just use this but do the same to the rest. This is lazy. You are literally copy pasting the same thing in all oracles you want to be penalizable.

That's not how programming works. Learn what inheritance and what mixins are.

You should create a super class for PenalizableOracles and add both this `punishment_info` there (rename to `penalty_info` and make it to a named tuple. Why the heck would you use a dict if it's just 2 attributes and you know all of them in advance?

And then implement this function there for all of them as `is_penalized()`.

Then for the oracles that can be penalizes they should also be inheriting from this mixin.